### PR TITLE
SignalR C++ tests passing and part of build

### PR DESCRIPTION
**Pull request by BrennanConroy**
Went through all failing tests and fixed them or removed them if they were no longer relevant.

Also added the client to the build. @halter73 and @mikaelm12 it might be a good idea to pull down the branch and run build.cmd from the signalr root to see if you have the required VS settings.

### CODE REVIEW

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/CommonLibTests.vcxproj [hunk 1]
```diff
@@ -193,5 +193,7 @@
       </AdditionalDependencies>
     </Lib>
   </ItemDefinitionGroup>
-  <Import Project=".\NativeTests.targets" />
+  <PropertyGroup>
+    <IsTestProject>true</IsTestProject>
```
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/CommonLibTests.vcxproj]: cc @pakrym

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/NativeTests.targets [hunk 2]
```diff
@@ -1,8 +0,0 @@
-<Project>
-  <PropertyGroup>
-    <VsTestConsole>$(VCIDEInstallDir)..\CommonExtensions\Microsoft\TestWindow\vstest.console.exe</VsTestConsole>
```
Comment (natemcmaster) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/NativeTests.targets]: Does this need to move into Cpp.Common.targets?

File: eng/targets/Cpp.Common.targets [hunk 3]
```diff
@@ -12,4 +12,7 @@
   <Target Name="Pack" />
   <Target Name="Restore" />
   <Target Name="ResolveNuGetPackageAssets" />
+  <Target Name="Test" DependsOnTargets="Build" Condition="'$(IsTestProject)' == 'true'" >
```
Comment (natemcmaster) [inline eng/targets/Cpp.Common.targets]:
```suggestion
  <Target Name="Test" Condition="'$(IsTestProject)' == 'true'" >
```
I'd recommend removing this target dependency so `build.cmd -nobuild -test` will work (cref #7137)

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/NativeTests.targets (hunk 2 above)
Comment (BrennanConroy) [reply src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/NativeTests.targets]: It wasn't being used. At least we don't think it was.

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/CommonLibTests.vcxproj (hunk 1 above)
Comment (pakrym) [reply src/Servers/IIS/AspNetCoreModuleV2/CommonLibTests/CommonLibTests.vcxproj]: 👍

**Review by natemcmaster — APPROVED** [review]

**Comment by halter73** [comment]
I ran "./build.cmd" in the src\SignalR directory, and it only compiled without running tests. I had to explicitly pass a "-test" flag to run the tests. Should we run tests by default now?

**Comment by BrennanConroy** [comment]
@natemcmaster It looks like the `<ImportGroup>` has run before the nuget packages are restored. Do you have any recommendations on how to fix that?
Line ref: https://github.com/aspnet/AspNetCore/pull/7160/files#diff-36a000fe432a66617010b714fa6a4d13L106

File: src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj [hunk 4]
```diff
@@ -17,16 +17,16 @@
   <Import Project="..\..\..\..\Build\Config.Definitions.props" />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v140</PlatformToolset>
```
Comment (jkotalik) [inline src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj]: Why downgrade from 141 to 140?

**Comment by natemcmaster** [comment]
> It looks like the <ImportGroup> has run before the nuget packages are restored. Do you have any recommendations on how to fix that?

The problem is your projects are using a really old version of NuGet and packages.config. It's been a while since I used them, but think these targets work under the expectation that you will have executed `NuGet.exe` on the .sln before you build. You might try looking into using newer version of NuGet and `PackageReference` instead.

File: src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj (hunk 4 above)
Comment (natemcmaster) [reply src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj]: Would be preferable to keep the toolset version consistent across all cpp projects in the repo. Maybe we should enforce this via eng/targets/Cpp.Common{props.targets} ?

Comment (BrennanConroy) [reply src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj]: Hopefully a temporary change. If you look at the C++ packages we restore, they are meant to run with v140. I also thought it was causing the build issues, turns out it was other things.

File: src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj [hunk 5]
```diff
@@ -17,16 +17,16 @@
   <Import Project="..\..\..\..\Build\Config.Definitions.props" />
```
Comment (jkotalik) [inline src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj]: Can you do me a big favor and make WindowsTargetPlatformVersion = 10.0.17134.0 everywhere (including here)? This PR I merged from 2.1 did some but I forgot to update more when merging from 2.1 to 2.2. https://github.com/aspnet/AspNetCore/commit/80bb14c3ef423cea4b970cc69ff273056c277428

Comment (jkotalik) [reply src/SignalR/clients/cpp/src/signalrclient/Build/VS/signalrclient.vcxproj]: You should just be able to find replace for 10.0.15063.0

File: src/SignalR/clients/cpp/test/signalrclienttests/internal_hub_proxy_tests.cpp [hunk 6]
```diff
@@ -65,37 +65,6 @@ TEST(on, cannot_register_handler_if_connection_not_in_disconnected_state)
     }
 }
 
-//TEST(invoke_event, invoke_event_invokes_event_and_passes_arguments)
```
Comment (halter73) [inline src/SignalR/clients/cpp/test/signalrclienttests/internal_hub_proxy_tests.cpp]: Why are we removing this test and the one below?

Comment (BrennanConroy) [reply src/SignalR/clients/cpp/test/signalrclienttests/internal_hub_proxy_tests.cpp]: invoke_event was some internal hub_proxy thing for invoking the `.on` handler of a hub_connection. We don't have hub_proxy.

Comment (halter73) [reply src/SignalR/clients/cpp/test/signalrclienttests/internal_hub_proxy_tests.cpp]: I see. You should probably rename this class to something other than internal_hub_proxy_tests.cpp. I'd also clean up signalrclient/CMakeLists.txt as it still includes hub_proxy.cpp and internal_hub_proxy.cpp.

Comment (halter73) [reply src/SignalR/clients/cpp/test/signalrclienttests/internal_hub_proxy_tests.cpp]: Also signalrclientdll.vcxproj.

**Comment by BrennanConroy** [comment]
Ping!

**Review by halter73 — APPROVED** [review]

**Pull request merged**